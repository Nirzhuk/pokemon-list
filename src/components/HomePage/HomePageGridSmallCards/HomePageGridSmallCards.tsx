import { useRouter } from 'next/router';
import React from 'react'
import styled from 'styled-components'
import capitalize from '../../../modules/app/helpers/formatters/capitalize';
import { APIURLReference, TextAligment } from '../../../modules/app/types';
import Title from '../../Common/Title';
import SmallCard from '../../SmallCard';

const HomePageGridSmallCardsContainer = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
`;

export interface HomePageGridSmallCardsProps {
    data: APIURLReference[]
}

const HomePageGridSmallCards = ({ data }: HomePageGridSmallCardsProps) => {
    const router = useRouter();

    const goToPokemonPageRoute = (pokemonName: string) => {
        router.push(`/${pokemonName}`)
    }

    if (data.length === 0) {
        return (<div>
            <Title as="h3" align={TextAligment.center} size="28px">There&apos; s not pokemons with that name, let&apos; s catch them ;)</Title>
        </div>)
    }
    return (
        <HomePageGridSmallCardsContainer>
            {data.map((i: APIURLReference) => <SmallCard onClick={() => goToPokemonPageRoute(i.name)} text={capitalize(i.name)} key={i.name} />)}
        </HomePageGridSmallCardsContainer>
    )
}

export default HomePageGridSmallCards