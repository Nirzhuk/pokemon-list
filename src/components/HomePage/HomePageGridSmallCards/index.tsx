export { default, type HomePageGridSmallCardsProps } from './HomePageGridSmallCards';