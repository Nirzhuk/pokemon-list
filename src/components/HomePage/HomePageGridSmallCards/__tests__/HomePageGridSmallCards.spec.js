import { render, screen } from '@testing-library/react'
import HomePageGridSmallCards from '../HomePageGridSmallCards'

describe('HomePageGridSmallCards', () => {
    const props = {
        data: [{
            name: "pikachu",
            url:"pokapi/pikachu"
        }]
    }
    const setup = (initialProps = props) => render(<HomePageGridSmallCards {...initialProps} />)
    
    it('Renders correctly', () => {
        const { container } = setup();

        expect(container).toMatchSnapshot();
    })

    it('Guards when gets 0 elements', () => {
        const { container } = setup({data: []});

        const text = "There' s not pokemons with that name, let' s catch them ;)"

        expect(screen.getByText(text)).toBeInTheDocument()
        expect(container).toMatchSnapshot();
    })
})