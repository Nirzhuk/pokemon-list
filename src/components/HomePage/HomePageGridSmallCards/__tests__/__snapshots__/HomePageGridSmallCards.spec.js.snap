// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageGridSmallCards Guards when gets 0 elements 1`] = `
<div>
  <div>
    <h3
      class="sc-bcXHqe kiCupq"
    >
      There' s not pokemons with that name, let' s catch them ;)
    </h3>
  </div>
</div>
`;

exports[`HomePageGridSmallCards Renders correctly 1`] = `
<div>
  <div
    class="sc-dkrFOg kieAM"
  >
    <div
      class="sc-gswNZR dgqJhH"
    >
      Pikachu
    </div>
  </div>
</div>
`;
